Mathlib/GroupTheory/Perm/ArrowAction.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2023 Junyan Xu, Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu, Antoine Chambert-Loir
+-/
+
+import Mathlib.GroupTheory.Subgroup.Basic
+import Mathlib.GroupTheory.GroupAction.Basic
+import Mathlib.Data.Set.Card
+
+/-!  Subgroup of `Equiv.Perm α` preserving a function `p : α → ι`
+
+Let `α` and `ι` by types.
+We consider `arrowAction : MulAction (Equiv.Perm α) (α → ι)`.
+
+* `arrowAction.mem_stabilizer_iff` proves that the stabilizer
+  of `Equiv.Perm α` of `p : α → ι` is the subgroup of `g : Equiv.Perm α`
+  such that `p ∘ g = p`.
+
+* `arrowAction.stabilizerMulEquiv` is the `MulEquiv` between this stabilizer and
+  the product, for `i : ι`, of `Equiv.Perm {a | p a = i}`.
+
+-/
+
+variable {α ι : Type*} {p : α → ι}
+
+open Equiv MulAction
+
+attribute [local instance] arrowAction
+
+namespace arrowAction
+
+lemma mem_stabilizer_iff {g : Perm α} :
+    g ∈ stabilizer (Perm α) p ↔ p ∘ g = p := by rw [eq_comm, ← g.comp_symm_eq]; rfl
```

Comment:
Arguably this should be
```lean
lemma mem_stabilizer_iff {g : DomMulAct (Perm α)} :
    g ∈ stabilizer (DomMulAct (Perm α)) p ↔ p ∘ (DomMulAct.mk.symm g :) = p := Iff.rfl
```
which doesn't need the "bad" arrowAction instance

Reply:
and the like for the whole file, I presume?
Should I put this in the DomMulAct namespace ?